src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs

```diff
@@ -1019,154 +1019,73 @@ public static bool IsEqual(this Version @this, Version other)
-        }
-
-        public static bool IsEqual(this ActiveDirectoryServerDownException @this, ActiveDirectoryServerDownException other)
-        {
-            // Name and ToString are different by design
-
-            return @this != null &&
-                other != null &&
-                // On full framework, line number may be method body start
-                // On Net Native we can't reflect on Exceptions and change its StackTrace
-                ((PlatformDetection.IsFullFramework || PlatformDetection.IsNetNative) ? true :
-                (@this.StackTrace == other.StackTrace)) &&
-                @this.Data.CheckSequenceEquals(other.Data) &&
-                @this.Source == other.Source &&
-                // On Net Native we can't reflect on Exceptions and change its HResult
-                (PlatformDetection.IsNetNative ? true : @this.HResult == other.HResult) &&
-                @this.HelpLink == other.HelpLink &&
-                CheckEquals(@this.InnerException, other.InnerException);
-        }
-
-        public static bool IsEqual(this SecurityException @this, SecurityException other)
-        {
-            // Name and ToString are different by design
-
-            return @this != null &&
-                other != null &&
-                // On full framework, line number may be method body start
-                // On Net Native we can't reflect on Exceptions and change its StackTrace
-                ((PlatformDetection.IsFullFramework || PlatformDetection.IsNetNative) ? true :
-                (@this.StackTrace == other.StackTrace)) &&
-                @this.Data.CheckSequenceEquals(other.Data) &&
-                @this.Source == other.Source &&
-                // On Net Native we can't reflect on Exceptions and change its HResult
-                (PlatformDetection.IsNetNative ? true : @this.HResult == other.HResult) &&
-                @this.HelpLink == other.HelpLink &&
-                CheckEquals(@this.InnerException, other.InnerException);
-        }
-
-        public static bool IsEqual(this NetworkInformationException @this, NetworkInformationException other)
-        {
-            // NetworkInformationException has a different StackTrace and HResult on Unix and OSX.
-
-            return @this != null &&
-                other != null &&
-                @this.Data.CheckSequenceEquals(other.Data) &&
-                // @this.Message == other.Message &&    // Message is platform dependent, as it is generated by the platform from the error code
-                @this.Source == other.Source &&
-                @this.HelpLink == other.HelpLink &&
-                CheckEquals(@this.InnerException, other.InnerException);
-        }
-
-        public static bool IsEqual(this SocketException @this, SocketException other)
-        {
-            // SocketException has a different StackTrace and HResult on Unix and OSX.
+            Assert.NotNull(@this);
```

Comment:
Are any of these IsEquals used when comparing Core with Core? BEcause I hope for sure that Message, etc matches in that situation.

Reply:
that's also for Core <--> Core. Some of the information gets lost during serialization because some ctors aren't passing the information correctly downwards. I will check which ones of those I'm talking about.

Reply:
It makes me nervous if we have no validation that a deserialized SqlException has any message at all, eg.

Reply:
then let's just pretend it hasn't been like this the entire time in master 🤕 (i'll fix it)